### PR TITLE
Feature/web3 384 publish nexeraidsiggating library in the ligo registry

### PR DESCRIPTION
**Pull request by frankhillard**

### CODE REVIEW

File: tezos-sig-gating-contracts/scripts/utils/helper.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+"use strict";
```
Comment (joelamouche) [inline tezos-sig-gating-contracts/scripts/utils/helper.js:1]: why do we have a .js file here 🤔 
You should probably ignore all js files in gitignore

File: tezos-sig-gating-contracts/tezos-lib-sig-gating-extendable/README.md [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+
+### Replay attack and expiration
+
+In order to prevent that a signature is used twice , the payload must incapsulate a "nonce" field.
+Once a user has received a signed message, he must consume the offchain signed message before an expiration (block level). The expiration has been introduced in case the user lose his authorization.
+
+### Forging a signature
+
+The request of a user (i.e. a contract invocation) is declared as a payload sent to a ReferenceSigner. The ReferenceSigner retrieves the nonce for the user (from contract storage), specifies the expiration (block level) and compact all using a keccak hashing and finally signs this "keccaked payload". This off-chain signature is produced by the ReferenceSigner.
+Then the produced signature and extra data (expiration, nonce, public key) are sent back to the user. The user can now send his transaction (which contains a `txAuthData` (payload + signature + public key + nonce + expiration)) to the smart contract that implements the signature verification mechanism.
```
Comment (joelamouche) [inline tezos-sig-gating-contracts/tezos-lib-sig-gating-extendable/README.md:17]: *signer public key

File: tezos-sig-gating-contracts/tezos-lib-sig-gating-extendable/README.md [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+
+![](./pictures/nexera%20global%20workflow.png)
+
+In this case the transaction fee is paid by the user.
+
+The transaction could also be executed by an operator on behalf of the user. In this case the transaction fee is paid by the operator.
+
+### off-chain signture workflow
+
+For a given calldata, the user receives from Nexera
```
Comment (joelamouche) [inline tezos-sig-gating-contracts/tezos-lib-sig-gating-extendable/README.md:27]: So the idea here is to write the general case, right? So maybe don't mention Nexera and say "The Signer" or "The Auth Provider" or "The KYC Provider"

File: tezos-sig-gating-contracts/tezos-lib-sig-gating-extendable/README.md [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+The request of a user (i.e. a contract invocation) is declared as a payload sent to a ReferenceSigner. The ReferenceSigner retrieves the nonce for the user (from contract storage), specifies the expiration (block level) and compact all using a keccak hashing and finally signs this "keccaked payload". This off-chain signature is produced by the ReferenceSigner.
+Then the produced signature and extra data (expiration, nonce, public key) are sent back to the user. The user can now send his transaction (which contains a `txAuthData` (payload + signature + public key + nonce + expiration)) to the smart contract that implements the signature verification mechanism.
+
+![](./pictures/nexera%20global%20workflow.png)
+
+In this case the transaction fee is paid by the user.
+
+The transaction could also be executed by an operator on behalf of the user. In this case the transaction fee is paid by the operator.
+
+### off-chain signture workflow
```
Comment (joelamouche) [inline tezos-sig-gating-contracts/tezos-lib-sig-gating-extendable/README.md:25]: *signature

File: tezos-sig-gating-contracts/tezos-lib-sig-gating-extendable/README.md [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+- nonce (custom nonce of the contract)
+- expiration (block level)
+- public key of Nexera signer role
+- signature of the payload hash
+
+<!-- ![](./pictures/nexera%20forge%20sig%20workflow.png) -->
+
+Then the user build the transaction and invoke the exec_gated_calldata entrypoint. In order to be able to verify an off-chain signed message, the exec_gated_calldata entrypoint expects a `txAuthData` parameter which contains the following fields
+
+- payload hash
```
Comment (joelamouche) [inline tezos-sig-gating-contracts/tezos-lib-sig-gating-extendable/README.md:39]: wait so the payload hash is still needed?

File: tezos-sig-gating-contracts/tezos-lib-sig-gating-extendable/README.md [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+- chain ID
+- user address
+- nonce
+- expiration (block level)
+- calldata (contract, entrypoint name, arguments)
+- public key
```
Comment (joelamouche) [inline tezos-sig-gating-contracts/tezos-lib-sig-gating-extendable/README.md:45]: *signer public key

**Review by joelamouche — COMMENTED** [review]
There is a lot of commented code that could be removed

File: tezos-sig-gating-contracts/scripts/utils/helper.js (hunk 1 above)
Comment (frankhillard) [reply tezos-sig-gating-contracts/scripts/utils/helper.js:1]: Yes I agree , this file should have been ignored

**Review by joelamouche — APPROVED** [review]
Looks good 🚀 

There is some leftover commented code here and there

**Pull request merged**